src/psyclone/tests/dynamo0p3_builtins_test.py

```diff
@@ -292,373 +58,392 @@ def test_dynbuiltfactory_str():
     assert "Factory for a call to a Dynamo built-in" in str(factory)
 
 
-def mesh_code_present(code):
-    '''this test checks for the existance of mesh code. This exists for
-    all builtins with dm = True (although it is not actually required!) so
-    each test can call this function'''
-    assert "      USE mesh_mod, ONLY: mesh_type" in code
-    assert "      TYPE(mesh_type), pointer :: mesh => null()" in code
-    output_dm_1 = (
-        "      !\n"
-        "      ! Create a mesh object\n"
-        "      !\n"
-        "      mesh => f1%get_mesh()\n"
-        "      !\n")
-    print output_dm_1
-    assert output_dm_1 in code
+# ------------- Adding (scaled) fields ------------------------------------- #
 
 
-def test_builtin_set_str():
-    ''' Check that the str method of DynSetFieldScalarKern returns the
+def test_X_plus_Y_str():    # Z = X + Y
```

Comment:
I realise that you've followed my (bad) example here in having a separate test for the str method of DynXPlusYKern and then another for its actual functionality. However, given that the setup of the two tests is identical I would drop this test and just include the check on `str(kern)` within the `test_X_plus_Y` test.

Reply:
Merged, as well as the rest of simple "one-operation"  built-ins tests.